cpp/pybind/visualization/gui/gui.cpp

```diff
@@ -693,6 +694,25 @@ void pybind_gui_classes(py::module &m) {
                  "it requires some internal setup in order to function "
                  "properly");
 
+    // ---- WidgetProxy ----
+    py::class_<WidgetProxy, UnownedPointer<WidgetProxy>, Widget> widgetProxy(
+            m, "WidgetProxy", "WidgetProxy");
+    widgetProxy.def(py::init<>(), "Creates a widget proxy")
+            .def("__repr__",
+                 [](const WidgetProxy &c) {
+                     std::stringstream s;
+                     s << "Proxy (" << c.GetFrame().x << ", " << c.GetFrame().y
+                       << "), " << c.GetFrame().width << " x "
+                       << c.GetFrame().height;
+                     return s.str();
+                 })
+            .def(
+                    "set_widget",
+                    [](WidgetProxy &w, UnownedPointer<Widget> proxy) {
+                        w.SetWidget(TakeOwnership<Widget>(proxy));
+                    },
+                    "Adds a proxy widget")
```

Comment:
Ensure documentation here is sufficient for a new user to correctly use this. This would at least need your PR description, and perhaps more detail. Also ensure C++ documentation is similar.

Reply:
pybind11 docs has been added, with similar content as CPP.